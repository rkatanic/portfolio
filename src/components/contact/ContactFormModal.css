.contact-form-modal {
  position: absolute;
  top: 50%;
  left: 50%;
  padding: 0.5rem;
  transform: translate(-50%, -50%) !important;
  margin: auto auto;
  background-color: var(--color-body);
  outline: none;
}

.contact-form-modal-content {
  padding: 1rem;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
  width: 100%;
}

.contact-form-modal-content-row {
  display: flex;
  align-items: center;
}

.contact-form-modal-content-icon {
  margin: 1rem;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 5rem;
  height: 5rem;
  background: var(--color-primary);
  font-size: 2rem;
  color: var(--color-title);
  font-weight: 600;
}

.contact-form-modal-content-icon .icon,
.contact-form-modal-content-icon .icon:hover {
  fill: var(--color-title);
  cursor: initial;
}

.contact-form-modal .icon-close {
  height: 1.75rem;
  float: right;
}

@media screen and (max-width: 767px) {
  .contact-form-modal {
    width: 100%;
    height: 100%;
  }

  .icon-close {
    margin: 1rem 1rem 0 0;
  }
}
