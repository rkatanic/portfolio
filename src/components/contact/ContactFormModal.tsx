import Modal from "react-modal";
import { ReactComponent as CloseIcon } from "../../assets/icons/close.svg";
import { ReactComponent as SuccessIcon } from "../../assets/icons/success.svg";

import "./ContactFormModal.css";

interface Props {
  isSuccess: boolean;
  isOpen: boolean;
  closeModal: () => void;
}

const ContactFormModal = ({
  isSuccess,
  isOpen,
  closeModal,
}: Props): JSX.Element => (
  <Modal
    style={{ overlay: { zIndex: "100", background: "rgb(0,0,0,0.5)" } }}
    className="contact-form-modal"
    ariaHideApp={false}
    isOpen={isOpen}
    onRequestClose={closeModal}
  >
    <CloseIcon onClick={closeModal} />
    <div className="contact-form-modal-content">
      {isSuccess ? (
        <>
          <SuccessIcon />
          <h3>Email successfully sent</h3>
          <p>Thanks for reaching out to me!</p>
          <p>I will respond as soon as possible.</p>
        </>
      ) : (
        <>
          <h3>Failed to send email</h3>
          <p>Oh no, something went wrong.</p>
          <p>Please try sending message again.</p>
        </>
      )}
    </div>
  </Modal>
);

export default ContactFormModal;
