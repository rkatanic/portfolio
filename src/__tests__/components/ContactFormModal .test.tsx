import { fireEvent, render } from "@testing-library/react";
import ContactFormModal from "../../components/contact/ContactFormModal";

describe("ContactFormModal ", (): void => {
  it("should render success message", (): void => {
    const { baseElement } = render(
      <ContactFormModal isSuccess={true} isOpen={true} closeModal={jest.fn()} />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("should render error message", (): void => {
    const { baseElement } = render(
      <ContactFormModal
        isSuccess={false}
        isOpen={true}
        closeModal={jest.fn()}
      />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it("should close modal", (): void => {
    const mockCloseModal = jest.fn();
    const { getByText } = render(
      <ContactFormModal
        isSuccess={false}
        isOpen={true}
        closeModal={mockCloseModal}
      />
    );

    fireEvent.click(getByText("close.svg"));

    expect(mockCloseModal).toHaveBeenCalledTimes(1);
  });
});
