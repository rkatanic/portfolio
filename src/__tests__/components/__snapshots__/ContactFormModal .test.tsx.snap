// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactFormModal  should render error message 1`] = `
<body
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay"
      style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(255, 255, 255, 0.75); z-index: 100;"
    >
      <div
        aria-modal="true"
        class="ReactModal__Content contact-form-modal"
        role="dialog"
        tabindex="-1"
      >
        <svg>
          close.svg
        </svg>
        <div
          class="contact-form-modal-content"
        >
          <div
            class="contact-form-modal-content-row"
          >
            <div
              class="contact-form-modal-content-icon"
            >
              !
            </div>
            <h3
              class="headline"
            >
              Sending 
              <br />
              email 
              <br />
              failed
            </h3>
          </div>
          <span>
            Oh no, something went wrong.
          </span>
          <span>
            Please try sending message later.
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ContactFormModal  should render success message 1`] = `
<body
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay"
      style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(255, 255, 255, 0.75); z-index: 100;"
    >
      <div
        aria-modal="true"
        class="ReactModal__Content contact-form-modal"
        role="dialog"
        tabindex="-1"
      >
        <svg>
          close.svg
        </svg>
        <div
          class="contact-form-modal-content"
        >
          <div
            class="contact-form-modal-content-row"
          >
            <div
              class="contact-form-modal-content-icon"
            >
              <svg>
                check.svg
              </svg>
            </div>
            <h3
              class="headline"
            >
              Email 
              <br />
              successfully 
              <br />
              sent
            </h3>
          </div>
          <span>
            Thanks for reaching me out!
          </span>
          <span>
            I will respond as soon as possible.
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
